builders/pallets-precompiles/precompiles/gmp.md

```diff
@@ -68,88 +68,153 @@ The following multilocation templates target accounts on other parachains with M
+              isEthereumStyle
+                ? { AccountKey20: { key: account } }
+                : { AccountId32: { id: account } },
+            ],
+          },
+        },
+      };
+
+      // Format multilocation object as a Polkadot.js type
+      const destination = registry.createType(
+        'VersionedMultiLocation',
+        versionedMultiLocation
+      );
+
+      // Wrap and format the multiLocation object into the precompile's input type
+      const userAction = new XcmRoutingUserAction({ destination });
+      const versionedUserAction = new VersionedUserAction({ V1: userAction });
+
+      // SCALE encode resultant precompile formatted objects
+      return versionedUserAction.toU8a();
+    }
+    ```
+
+=== "V2"
+
+    ```javascript
+    import { TypeRegistry, Enum, Struct } from '@polkadot/types';
 
-  // Wrap and format the multiLocation object into the precompile's input type
-  const userAction = new XcmRoutingUserAction({ destination });
-  const versionedUserAction = new VersionedUserAction({ V1: userAction });
+    // Creates a type registry to properly work with the precompile's input types
+    const registry = new TypeRegistry();
 
-  // SCALE encode resultant precompile formatted objects
-  return versionedUserAction.toU8a();
-}
-```
+    // Define the precompile's input types VersionedUserAction and XcmRoutingUserActionWithFee
+    class VersionedUserAction extends Enum {
+      constructor(value) {
+        super(registry, { V2: XcmRoutingUserActionWithFee }, value);
+      }
+    }
+    class XcmRoutingUserActionWithFee extends Struct {
+      constructor(value) {
+        super(
+          registry,
+          { destination: 'VersionedMultiLocation', fee: 'u256' },
+          value
+        );
+      }
+    }
+
+    // A function that creates a SCALE encoded payload to use with transferTokensWithPayload
+    function createMRLPayload(parachainId, account, isEthereumStyle, fee) {
+      // Create a multilocation object based on the target parachain's account type
+      const versionedMultiLocation = {
+        v1: {
+          parents: 1,
+          interior: {
+            X2: [
+              { Parachain: parachainId },
+              isEthereumStyle
+                ? { AccountKey20: { key: account } }
+                : { AccountId32: { id: account } },
+            ],
+          },
+        },
+      };
+
+      // Format multilocation object as a Polkadot.js type
+      const destination = registry.createType(
+        'VersionedMultiLocation',
+        versionedMultiLocation
+      );
+
+      // Wrap and format the multiLocation object into the precompile's input type
+      const userAction = new XcmRoutingUserActionWithFee({ destination, fee });
+      const versionedUserAction = new VersionedUserAction({ V2: userAction });
+
+      // SCALE encode resultant precompile formatted objects
+      return versionedUserAction.toU8a();
+    }
+    ```
 
 ## Restrictions {: #restrictions }
 
```

Comment:
On line 225: I would add Errors in constructing a multilocation will result in reverts, which will trap tokens and **result in** a loss of funds